lib/crawlers/base_crawler.rb

```diff
@@ -0,0 +1,153 @@
+require_relative "../priority_spidr/priority_spidr"
+require "cgi"
+require "pathname"
+require "csv"
+
+module Crawlers
+  # Base class for crawlers
+  class BaseCrawler
+    # Array for storing whatever the crawler is looking for
+    # @return [Array]
+    attr_reader :hits
+
+    # Crawl limiters
+
+    # @return [Integer, NilClass]
+    attr_accessor :max_crawl
+    # @return [Float, NilClass]
+    attr_accessor :max_time
+    # @return [Integer, NilClass]
+    attr_accessor :max_hits
+
+    # @param url [String]
+    #
+    # @param stubs_path [Pathname, NilClass]
+    def initialize(url, stubs_path = nil)
+      @starting_url = url
+      @hits = []
+      if stubs_path.nil?
```

Comment:
Think this can be made slightly shorter?

Reply:
The initialize function? I could wrap the check into `load_stubs` and remove the warning print

Reply:
``` ruby
# @param url [String]
#
# @param stubs_path [Pathname, NilClass]
def initialize(url, stubs_path = nil)
  @starting_url = url
  @hits = []
  @priority_stubs = load_stubs(stubs_path)
end

private

# Load priority stubs to use.
# Expects file format to be plain text, one stub per line.
# If `path` is `nil`, an empty array will be returned.
#
# @param path [String]
#
# @return [Array<String>]
def load_stubs(path)
  return [] if stubs_path.nil?

  return File.new(path).readlines.map(&:chomp)
end
```